temporalcli/commandsmd/commands.yml

```diff
@@ -0,0 +1,3023 @@
+
+# A Command entry uses the following format:
+# - name: The full command path with parent commands. (string)
+#   summary: A concise command explanation. (string)
+#   description: A detailed command explanation. (string)
+#   has-init: invokes `initCommand` method. (bool)
+#   exact-args: Require this exact number of args. (int)
+#   maximum-args: Require this maximum number of args. (int)
+#   ignores-missing-env: Ignore missing environment variables. (bool)
+#   options: A list of options. (Option[])
+#     - name: The option name. (string)
+#       type: The option type. (string)
+#       description: The option description. (string)
+#       required: Whether the option is required. (bool)
+#       short: The single letter short version of name (i.e. a for address). (string)
+#       env: Binds the environment variable to this flag. (string)
+#       default: The default value. No default means zero value of the type. (string)
+#       enum-values: A list of possible values for the string-enum type. (string[])
+#       aliases: A list of aliases for the option. (string[])
+#   option-sets: A list of option sets. (string[])
+
+# * name, summary, and descrption are required fields. All other fields are optional.
+# * Available option types are `bool`, `duration`, `int`, `string`, `string[]`, 
+#   `string-enum`, or `timestamp`.
+
+# OPTION SET OVERVIEW
+
+# An options set declaration is the equivalent of pasting those options into the 
+# bulleted options list.
+
+# - name: The name of the option set. (string)
+#   options: A list of options. Same template as Command options above (Option[])
+
+# Options that are similar but slightly different don't need to be in option sets.
+# Reserve option sets for when the behavior of the option is the same across commands.
+
+commands:
+  - name: temporal
+    summary: Temporal command-line interface and development server
+    description: |
+      The Temporal CLI manages, monitors, and debugs Temporal apps. It lets you run
+      a local Temporal Service, start Workflow Executions, pass messages to running
+      Workflows, inspect state, and more.
+
+      * Start a local development service:
+            `temporal server start-dev`
+      * View help: pass `--help` to any command:
+            `temporal activity complete --help`
+    has-init: true
+    options:
+      - name: env
+        type: string
+        description: Active environment name (`ENV`).
+        default: default
+        env: TEMPORAL_ENV
+      - name: env-file
+        type: string
+        description: |
+          Path to environment settings file.
+          (defaults to `$HOME/.config/temporalio/temporal.yaml`).
+      - name: log-level
+        type: string-enum
+        enum-values:
+          - debug 
+          - info
+          - warn
+          - error
+          - never
+        description: |
+          Log level.
+          Default is "info" for most commands and "warn" for `server start-dev`.
```

Comment:
No, I want to get rid of this eventually and use a proper Markdown renderer for both the description and flags. The reason it was needed for flags is because newlines show up in the generated output, so having newlines inserted in (to-the-user) random places really doesn't look good. I think Andrew has since fixed this? In which case I have no opinion.

Reply:
I think we should have a general/consistent rule for Markdown `description`. Right now it seems a bit inconsistent.

Reply:
Agree, eventually; we can make this consistent once we have a proper Markdown formatter IMO.